nerdlandbot/scheduler/PurgeScheduler.py

```diff
@@ -14,17 +14,19 @@ async def purge_messages(bot):
     info("Purging messages")
     guilds_data = await get_all_guilds_data()
     for guild_data in guilds_data:
-        for text_channel, max_age in guild_data.purgers.items():
-            before = datetime.today() - timedelta(days=max_age)
-            channel = bot.get_channel(int(text_channel))
-            try:
-                if channel:
-                    await channel.purge(check=check, before=before)
-            except:
-                fatal(
-                    f"Failed to purge messages for channel {channel.name} in guild {guild_data.guild_id}. Does the bot have appropriate permissions on that channel?"
-                )
-
-
+        if bot.is_purging[str(guild_data.guild_id)] == False:
```

Comment:
Technically this could give issues if multiple threads execute this code for the same guild, as the second thread could evaluate the if condition before the first thread set the flag.

This seems like an edge case that is highly unlikely to occur, so i doubt it's really necessary to lock it?
@DrSkunk @Autom3 might have an opinion on this?

Reply:
As it's an edge case that's unlikely to occur, it's not necessary for me to resolve.